manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat' => [Package['snmp-client'], Package[$snmp::params::package_name]],
```

Comment:
it's not hardcoded. 'snmpd' is just the name of the resource, the real package name given with the `name` parameter:
```
  package { 'snmpd':
    ensure => $package_ensure,
    name   => $package_name,
  }
```